pkg/controller/handlers/mcpcatalog/github.go

```diff
@@ -53,29 +159,59 @@ func readGitHubCatalog(catalogURL string) ([]types.MCPServerCatalogEntryManifest
 	branch := "main"
 	if len(parts) > 2 {
 		branch = parts[2]
```

Comment:
Now, that we are correctly allowing branch names to have `/` in them, I think this needs to change, too.